### PR TITLE
modal blocked

### PR DESCRIPTION
**Pull request by juangdelvalle**
Set a presentationController delegate to be able to capture the event when the presented view is dismissed by a swap down with the new modality management of iOS 13, implementing the presentationControllerDidDismiss method

### CODE REVIEW

File: Yoshi.podspec [hunk 1]
```diff
@@ -7,26 +7,18 @@
...
-  s.summary          = "A helpful companion for your iOS app."
-
-# This description is used to generate tags and improve search results.
-#   * Think: What does it do? Why did you write it? What is the focus?
-#   * Try to keep it short, snappy and to the point.
-#   * Write the description between the DESC delimiters below.
-#   * Finally, don't worry about the indent, CocoaPods strips it!
-  s.description      = <<-DESC
-		       Yoshi is a convenient wrapper around the UI code that is often needed for displaying debug menus. Out of the box, Yoshi provides easy-to-implement date, list and custom menus.
-                       DESC
```
Comment (MichaelCampbell) [inline Yoshi.podspec]: Is the description no longer required — curious why you removed it?

**Comment by juangdelvalle** [comment]
@paulmiard @MichaelCampbell I couldn't add you as a reviewer on the PR, so if you can add it to your list of pending reviews to check it when you have a chance it would be great. Thanks on advance!

File: Yoshi.podspec [hunk 2]
```diff
@@ -7,26 +7,18 @@
...
-
-  s.homepage         = "https://github.com/prolificinteractive/Yoshi"
-  s.screenshots     = "https://raw.githubusercontent.com/prolificinteractive/Yoshi/a6e85e87cbd67f2bb3bfe60157e7b13281d80f20/Images/Yoshi.png", "https://raw.githubusercontent.com/prolificinteractive/Yoshi/c66cdf8dc2ab643fe57996d20d3cd37b8b70ceff/Images/Yoshi_iPad.png"
-  s.license          = { :type => 'MIT', :file => 'LICENSE' }
-  s.author           = { "Michael Campbell" => "Michael@prolificinteractive.com" , "Luna An" => "luna@prolificinteractive.com"}
-  s.source           = { :git => "https://github.com/prolificinteractive/Yoshi.git", :tag => s.version.to_s }
-
-  s.platform     = :ios, '8.0'
```
Comment (MichaelCampbell) [inline Yoshi.podspec:29]: Is platform no longer required?

File: Yoshi.podspec [hunk 3]
```diff
@@ -7,26 +7,18 @@
...
+  s.name             = 'Yoshi'
+  s.version          = '3.0.1'
+  s.summary          = 'A helpful companion for your iOS app.'
+  s.license          = 'MIT'
```
Comment (MichaelCampbell) [inline Yoshi.podspec:13]: Should we no longer link reference the license file?

**Review by MichaelCampbell — APPROVED** [review]
This looks great, thanks for working on it @juangdelvalle. Sorry for the delay on reviewing.

I left some inline comments, but I think we should be good to merge once the travis build passes. Let me know if you need some support getting it to pass.

File: Yoshi.podspec [hunk 4]
```diff
@@ -7,26 +7,18 @@
...
+  s.homepage         = 'https://github.com/prolificinteractive/Yoshi'
+  s.authors          = { 'Michael Campbell' => 'Michael@prolificinteractive.com' , 'Luna An' => 'luna@prolificinteractive.com'}
+  s.source           = { :git => "https://github.com/prolificinteractive/Yoshi.git", :tag => s.version }
+  s.screenshots      = 'https://raw.githubusercontent.com/prolificinteractive/Yoshi/a6e85e87cbd67f2bb3bfe60157e7b13281d80f20/Images/Yoshi.png", "https://raw.githubusercontent.com/prolificinteractive/Yoshi/c66cdf8dc2ab643fe57996d20d3cd37b8b70ceff/Images/Yoshi_iPad.png'
+
+  s.ios.deployment_target = '8.0'
+  s.swift_versions   = ['4.2', '5.0', '5.1']
```
Comment (paulmiard) [inline Yoshi.podspec]: There is an open PR that takes care of Swift 5 changes. Don't we need to merge it before adding 5.x to the list?
See https://github.com/prolificinteractive/Yoshi/pull/83

File: Yoshi.podspec (hunk 3 above)
Comment (juangdelvalle) [reply Yoshi.podspec:13]: as I understand, is no longer required to make a reference to a license file, but I can change it to make the reference anyway.

File: Yoshi.podspec (hunk 1 above)
Comment (juangdelvalle) [reply Yoshi.podspec]: it was by mistake, I'll add it back right away

File: Yoshi.podspec (hunk 2 above)
Comment (juangdelvalle) [reply Yoshi.podspec:29]: platform is now specified in  `s.ios.deployment_target = '8.0'`

File: Yoshi.podspec (hunk 4 above)
Comment (juangdelvalle) [reply Yoshi.podspec]: In this PR I also migrated the code to Swift 5 (the other PR wasn't approved yet so I thought about killing two birds with the same stone)

**Comment by juangdelvalle** [comment]
> This looks great, thanks for working on it @juangdelvalle. Sorry for the delay on reviewing.
> 
> I left some inline comments, but I think we should be good to merge once the travis build passes. Let me know if you need some support getting it to pass.

@MichaelCampbell I might need help to get it to pass, the error that I'm getting is: `SWIFT_VERSION '5.0' is unsupported, supported versions are: 3.0, 4.0, 4.2. (in target 'YoshiTests')` I wonder if the supported swift version is something that must be setted on Travis

File: Yoshi.podspec (hunk 4 above)
Comment (paulmiard) [reply Yoshi.podspec]: I didn't look at the details in the other PR but looked like there was more going on. Are you saying the other PR can be declined all together once this is merged? I guess I will diff it to see what else was in there once we merge but happy to hear your thoughts

**Comment by MichaelCampbell** [comment]
> @MichaelCampbell I might need help to get it to pass, the error that I'm getting is: `SWIFT_VERSION '5.0' is unsupported, supported versions are: 3.0, 4.0, 4.2. (in target 'YoshiTests')` I wonder if the supported swift version is something that must be setted on Travis

I took a look into this framework and it seems that they're doing something similar to us: https://github.com/DianQK/TransitionTreasury/blob/master/.travis.yml

Maybe try bumping the pod version to latest @juangdelvalle

**Comment by Armenm** [comment]
Hi @juangdelvalle, any progress on this PR?

**Pull request closed without merging**